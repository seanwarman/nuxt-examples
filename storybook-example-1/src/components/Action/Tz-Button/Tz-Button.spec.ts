/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper } from '@vue/test-utils';
import TzButton from './Tz-Button.vue';
import { expect, describe, beforeEach, it } from 'vitest';

const text = "I'm a button";
let wrapper: VueWrapper;

describe('Button setup', () => {
  beforeEach(() => {
    wrapper = mount(TzButton, {
      props: {
        iconPosition: 'before'
      },
      slots: {
        default: text
      }
    });
  });
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('renders as button element', () => {
    expect(wrapper.html()).toContain('button');
  });
  it('contains the provided text', () => {
    expect(wrapper.text()).toContain(text);
  });
  it("emits a 'click' event when it is clicked", () => {
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted()).toHaveProperty('click');
  });
});
