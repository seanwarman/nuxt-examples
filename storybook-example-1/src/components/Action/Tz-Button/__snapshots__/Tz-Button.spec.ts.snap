// Vitest Snapshot v1

exports[`Button setup > renders correctly 1`] = `
<div
  data-v-app=""
>
  <button
    class="flex items-center justify-center gap-2 text-sm font-extrabold uppercase tracking-wider focus:ring bg-dark hover:bg-black text-white ring-dark px-8 py-3 rounded-[3px]"
  >
    
    
    
    I'm a button
    
  </button>
</div>
`;
